macros/printing.cfg

```diff
@@ -0,0 +1,179 @@
+    {% set unretract_start = params.UNRETRACT_START|default(9)|int %}
+    {% set unretract_speed = params.UNRETRACT_SPEED|default(1500)|int %}
+
+    RESPOND MSG='Print Start'
+    M117 Homing
+    G28
+
+    PREHEAT BED={target_bed} CHAMBER={target_chamber} FILAMENT={filament_type}
+    CALIBRATE SIZE={fl_size}
+
+    M117 Heating ~bed~{target_bed}~degrees~
+    M190 S{target_bed} # Set bed to final temp and wait
+
+    M117 Heating ~extruder~: {target_extruder}~degrees~
+    M109 S{target_extruder} # Set extruder to final temp and wait
+
+    G1 Z5 F5000 # Move head 5mm from bed surface
+    G90 # Absolute Positioning
+    M83 # Make the E relative independant of other axis
+    G1 E{unretract_start} F{unretract_speed} # Unretract filament
+    G92 E0 # reset extruder
+
+    RESPOND MSG='Printing'
+    M117 Printing
+
+[gcode_macro PREHEAT]
+description: Preheat the printer bed/extruder and wait for chamber temps if requested.
+gcode:
+    # command params
+    {% set target_bed = params.BED|default(110)|int %}
+    {% set target_extruder_hold = params.EXTRUDER_HOLD|default(150)|int %}
+    {% set target_chamber = params.CHAMBER|default(0)|int %}
+    {% set filament_type = params.FILAMENT|default("ABS")|upper %}
+    {% set park_speed = params.PARK_SPEED|default(18000)|int %}
+    {% set min_soak_time = params.MIN_SOAK_TIME|default(0)|int %}
+    {% set string_filament_types = params.NEVERMORE_FILAMENT_TYPES|default('ABS,ASA')|upper|replace(" ", "") %}
+    # variables
+    {% set filament_types = string_filament_types.split(',') %}
+
+    # Chamber/time soak for some filament types
+    {% if filament_type in filament_types %}
+        RESPOND MSG='Preheat Start'
+        M117 Preheat Start
+        G90 # absolute positioning
+        PARK_BED
+
+        M104 S{target_extruder_hold} # Set extruder to hold temp
+        M106 S255 # Turn on Part Cooling Fan to 100%
+
+        M117 Heating ~bed~{target_bed}~degrees~
+        M190 S{target_bed} # Set bed to final temp and wait
+
+        _CIRCULATION_START FILAMENT={filament_type}
+
+        # Timed bed soak
+        {% if min_soak_time > 0 %}
+            {% for timer in range(min_soak_time|int,0,-1) %}
+                M117 Soaking ~bed~ {timer|int}m
+                M105
+                G4 P60000
+            {% endfor %}
+        {% endif %}
+
+        # Chamber soak
+        {% if printer['temperature_sensor chamber'] and target_chamber > 0 and
+            printer['temperature_sensor chamber'].temperature < target_chamber %}
+            M117 Soaking ~chamber~ {target_chamber}~degrees~
+            TEMPERATURE_WAIT SENSOR="temperature_sensor chamber" MINIMUM={target_chamber}
+        {% endif %}
+
+        M107 # Stop part cooling fan
+        RESPOND MSG='Preheat Complete'
+        M117 Preheat Complete
+    {% endif %}
+
+[gcode_macro PRINT_END]
+description: Common print end macro.
+gcode:
+    # command params
+    {% set filament_type = params.FILAMENT|default("ABS")|upper %}
```

Comment:
default this to PLA, that will make cancel skip cooldown. if we even have such a thing here yet